src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -732,51 +732,11 @@ internal Result CreateResult(ResultVersionOne v1Result)
                     result.AnalysisTarget = CreateFileLocation(v1Result.Locations[0].AnalysisTarget);
                 }
 
-                if (v1Result.RuleKey == null)
-                {
-                    result.RuleId = v1Result.RuleId;
-                }
-                else
-                {
-                    if (v1Result.RuleId == null)
-                    {
-                        result.RuleId = v1Result.RuleKey;
```

Comment:
>result [](start = 24, length = 6)

This was never right. It was never valid to assign a `ruleKey` to a `ruleId`. The situation "`ruleId == null; ruleKey != null`" meant "This rule has no id (which is perfectly valid, both in v1 and v2). If you want to know more about this rule (like its description), you can find the rule metadata here."